lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -37,7 +37,7 @@ class ProxyGenerator
      * Used to match very simple id methods that don't need
      * to be decorated since the identifier is known.
      */
-    const PATTERN_MATCH_ID_METHOD = '((public\s+)?(function\s+%s\s*\(\)\s*)\s*{\s*return\s*\$this->%s;\s*})i';
+    const PATTERN_MATCH_ID_METHOD = '((public\s+)?(function\s+%s\s*\(\)\s*)\s*(?::\s*[\w\\\\]+\s*)?{\s*return\s*\$this->%s;\s*})i';
```

Comment:
php type identifiers follow the regex `'/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]+$/'`